presto-spark/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -194,31 +194,33 @@ private PrestoQueryExecution(
         @Override
         public List<List<Object>> execute()
         {
-            List<Tuple2<Integer, byte[]>> collectedRdd = rdd.collect();
+            List<Tuple2<Integer, byte[]>> collectedRdd;
+            try {
+                collectedRdd = rdd.collect();
+                commit();
+            }
+            catch (RuntimeException executionFailure) {
+                try {
+                    rollback();
+                }
+                catch (RuntimeException rollbackFailure) {
+                    if (executionFailure != rollbackFailure) {
```

Comment:
How come these two exceptions be the same??

Reply:
For example when transaction manager starts throwing the same exception on both, `commit` and `rollback` due to some weird internal state (e.g.: transaction is gone?). 

The `addSupressed` method doesn't allow cyclic references. It throws an `IllegalArgumentException` when self suppressing occurs.